core/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsResponse.java

```diff
@@ -80,76 +81,43 @@ public ClusterStatsIndices getIndicesStats() {
     }
 
     @Override
-    public ClusterStatsNodeResponse[] getNodes() {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public Map<String, ClusterStatsNodeResponse> getNodesMap() {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public ClusterStatsNodeResponse getAt(int position) {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public Iterator<ClusterStatsNodeResponse> iterator() {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
         timestamp = in.readVLong();
-        status = null;
-        if (in.readBoolean()) {
-            // it may be that the master switched on us while doing the operation. In this case the status may be null.
-            status = ClusterHealthStatus.fromValue(in.readByte());
-        }
         clusterUUID = in.readString();
-        nodesStats = new ClusterStatsNodes(in);
-        indicesStats = ClusterStatsIndices.readIndicesStats(in);
+        // it may be that the master switched on us while doing the operation. In this case the status may be null.
+        status = in.readOptionalWriteable(ClusterHealthStatus::readFrom);
+        nodes = in.readList(ClusterStatsNodeResponse::readNodeResponse);
+
+        // built from nodes rather than from the stream directly
+        nodesStats = new ClusterStatsNodes(nodes);
```

Comment:
Nope. Removed it from both (worse: one was `Streamable` and one was `Writeable`)!